webclient/src/components/Toast/reducer.ts

```diff
@@ -6,41 +6,54 @@ export const ACTIONS = {
 }
 
 export const initialState = {
-  toasts: new Map()
+  toasts: {}
 }
 
-export function reducer(state, action) {
-  const { type, payload } = action
+export function reducer(state, { type, payload }) {
+  const { key, children } = payload;
+
   switch (type) {
     case ACTIONS.ADD_TOAST: {
-      const newState = { ...state }
-      newState.toasts = new Map(Array.from(state.toasts))
-      const { toasts } = newState;
-      const { key, children } = payload
-      toasts.set(key, { isOpen: false, children })
-      return newState
+      return {
+        ...state,
+        toasts: {
+          ...state.toasts,
+          [key]: {
+            isOpen: false,
+            children,
+          },
+        },
+      };
```

Comment:
This is a great enhancement to the overall readability and simplicity of this reducer.  Nice work!

Something to consider is introducing the canonical [immutability-helper](https://github.com/kolodny/immutability-helper).  There's a mini-DSL to learn, but it will make short work of these state updates in the future, and remove the need for us to manually enforce immutability.

Reply:
Good idea, I've created a tech debt ticket to look into that.

https://trello.com/c/kyNW0P7x/16-look-into-using-immutability-helper-for-redux-mutations